app/controllers/cart_controller.rb

```diff
@@ -1,4 +1,6 @@
 class CartController < ApplicationController
+  before_action :require_all_but_admin
```

Comment:
This works but maybe lets change this to allow_all_but_admin so it is a little clearer